e2e/utils.go

```diff
@@ -473,16 +473,161 @@ func (s SimplyBlock) getStoragenode() (string, error) {
 
 	// TODO: get a random storage node
 	storageNodes, ok := out.([]interface{})[0].(map[string]interface{})
+
 	if !ok {
 		return "", errors.New("failed to get storage node from simplyblock api")
 	}
 	sn, ok := storageNodes["hostname"].(string)
+	fmt.Println("^^^^^^")
+	fmt.Println(sn)
+	fmt.Println("^^^^^^")
+	if !ok {
+		return "", errors.New("failed to get storage node from simplyblock api")
+	}
+	return sn, nil
+}
+
+func (s SimplyBlock) numberOfNodes() (int, error) {
+	var rpcClient util.RPCClient
+	rpcClient.ClusterID = s.UUID
+	rpcClient.ClusterIP = s.IP
+	rpcClient.ClusterSecret = s.Secret
+
+	rpcClient.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+
+	out, err := rpcClient.CallSBCLI("GET", "/storagenode", nil)
+	if err != nil {
+		return 0, err
+	}
+
+	//get the number of storage nodes
+	sn := len(out.([]interface{}))
+	return sn, nil
+
+}
+func (s SimplyBlock) getStoragenodeId(random int) (string, error) {
+	var rpcClient util.RPCClient
+	rpcClient.ClusterID = s.UUID
+	rpcClient.ClusterIP = s.IP
+	rpcClient.ClusterSecret = s.Secret
+
+	rpcClient.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+
+	// get the list of storage nodes
+	out, err := rpcClient.CallSBCLI("GET", "/storagenode", nil)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: get a random storage node
+
+	storageNodes, ok := out.([]interface{})[random].(map[string]interface{})
+
+	if !ok {
+		return "", errors.New("failed to get storage node from simplyblock api")
+	}
+	sn, ok := storageNodes["id"].(string)
```

Comment:
please use the `uuid` instead of the `id`

Reply:
done